alica_engine/src/engine/teammanager/TeamManager.cpp

```diff
@@ -138,49 +175,138 @@ bool TeamManager::isAgentActive(AgentIDConstPtr agentId) const
  */
 bool TeamManager::isAgentIgnored(AgentIDConstPtr agentId) const
 {
-    auto agentEntry = _agents.find(agentId);
-    if (agentEntry != _agents.end()) {
-        return agentEntry->second->isIgnored();
-    } else {
-        return false;
+    Agent* agent = getAgent(agentId);
+    if (agent) {
+        agent->isIgnored();
     }
+    return false;
 }
 
 void TeamManager::setAgentIgnored(AgentIDConstPtr agentId, const bool ignored) const
 {
-    auto agentEntry = _agents.find(agentId);
-    if (agentEntry != _agents.end()) {
-        agentEntry->second->setIgnored(ignored);
+    Agent* agent = getAgent(agentId);
+    if (agent) {
+        agent->setIgnored(ignored);
     }
 }
 
 bool TeamManager::setSuccess(AgentIDConstPtr agentId, const AbstractPlan* plan, const EntryPoint* entryPoint)
 {
-    auto agentEntry = _agents.find(agentId);
-    if (agentEntry != _agents.end()) {
-        agentEntry->second->setSuccess(plan, entryPoint);
-        return true;
+    Agent* agent = getAgent(agentId);
+    if (agent) {
+        agent->setSuccess(plan, entryPoint);
     }
     return false;
 }
 
 bool TeamManager::setSuccessMarks(AgentIDConstPtr agentId, const IdGrp& suceededEps)
 {
-    auto agentEntry = _agents.find(agentId);
-    if (agentEntry != _agents.end()) {
-        agentEntry->second->setSuccessMarks(suceededEps);
-        return true;
+    Agent* agent = getAgent(agentId);
+    if (agent) {
+        agent->setSuccessMarks(suceededEps);
     }
     return false;
 }
 
 const DomainVariable* TeamManager::getDomainVariable(AgentIDConstPtr agentId, const std::string& sort) const
 {
-    auto agentEntry = _agents.find(agentId);
-    if (agentEntry != _agents.end()) {
-        return agentEntry->second->getDomainVariable(sort);
+    Agent* agent = getAgent(agentId);
+    if (agent) {
+        return agent->getDomainVariable(sort);
     }
     return nullptr;
 }
 
+AgentGrp TeamManager::updateAgents(bool changedSomeAgent)
+{
+    AgentsCache::AgentMap& agents = *_agentsCache.get();
+    changedSomeAgent = false;
+    AgentGrp deactivatedAgentIds;
+    for (const auto& agent : agents) {
```

Comment:
please use the type of agent instead of auto here

Reply:
convention justifies using auto for this kind of container iteration.